### ISSUE TITLE
loss is same for each tpu core when using seed

### ISSUE THREAD

**Issue by gakshaygupta** [pytorch/xla#1797]
## ❓ Questions and Help
But when using distributed sampler to overcome this problem the IO increase by a factor of 5 even after increasing num_workers in dataloader.

**Comment by dlibenzi** [comment]
Can you try to explain better the issue?
Maybe with some code too?

**Comment by gakshaygupta** [comment]
when torch.manual_seed() along with a dataloader with random sampling is used in the training the error for each core of the tpu is same. This might be because manual_seed defined the way the dataloader sampled the data for each core which is same because of manual_seed which causes the error to be the same. Also when not using manual_seed the dataloader will not sample identical samples for each core but they might be overlapping which is a problem. So then I used a distributed sampler but the problem is that even after increasing the num_workers in data loader, time taken for IO is even more compared to that of dataloader with random sampling.

**Comment by taylanbil** [comment]
In order to guarantee same model weights per core at initialization, it is highly recommended that manual_seed is set. [Here](https://github.com/pytorch/xla/blob/master/test/test_train_mp_mnist.py#L88) is an example of how to use distributed sampler correctly in multiprocessing setting.

In order to diagnose the increase in IO, it may be useful to see the code.

**Comment by gakshaygupta** [comment]
here is the code https://github.com/gakshaygupta/Stereo-disparity and the parameters that I used to run the model : are batch_size=24 iterations=1000 log_interval=10 num_workers=2.

**Comment by gakshaygupta** [comment]
> In order to guarantee same model weights per core at initialization, it is highly recommended that manual_seed is set. [Here](https://github.com/pytorch/xla/blob/master/test/test_train_mp_mnist.py#L88) is an example of how to use distributed sampler correctly in multiprocessing setting.
> 
> In order to diagnose the increase in IO, it may be useful to see the code.

I was following the mentioned code for reference.

**Comment by gakshaygupta** [comment]
any updates?
